hadoop-yarn-project/hadoop-yarn/hadoop-yarn-csi/src/test/java/org/apache/hadoop/yarn/csi/client/TestCsiClient.java

```diff
@@ -36,23 +34,24 @@
  */
 public class TestCsiClient {
 
-  private static File testRoot = null;
+  private static File socketFile = null;
   private static String domainSocket = null;
   private static FakeCsiDriver driver = null;
 
   @BeforeClass
   public static void setUp() throws IOException {
-    testRoot = GenericTestUtils.getTestDir("csi-test");
-    File socketPath = new File(testRoot, "csi.sock");
-    FileUtils.forceMkdirParent(socketPath);
-    domainSocket = "unix://" + socketPath.getAbsolutePath();
+    socketFile = new File("/tmp", "yarn-csi-test.sock");
+    domainSocket = "unix://" + socketFile.getAbsolutePath();
     driver = new FakeCsiDriver(domainSocket);
   }
 
   @AfterClass
   public static void tearDown() throws IOException {
-    if (testRoot != null) {
-      FileUtils.deleteDirectory(testRoot);
+    if (socketFile != null && socketFile.exists()) {
+      if (!socketFile.delete()) {
+        String message = "Unable to delete file " + socketFile + ".";
+        throw new IOException(message);
+      }
```

Comment:
This removal seems quite weird, throwing IOE and checking this much. I checked other tests, most of them do like :
    new File(socketFile).delete(); or FileUtils.delete(..), you can check TestConfiguration.java for one such reference

Reply:
Hi @ayushtkn,
Thanks for your review, here I wanted to keep the logical as before(the FileUtils.deleteDirectory method) :) done